catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/SearchFormsSymbolsIT.java

```diff
@@ -0,0 +1,438 @@
+import ddf.catalog.operation.impl.CreateResponseImpl;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.QueryResponseImpl;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.security.Subject;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.SocketAddress;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import javax.xml.bind.JAXBException;
+import javax.xml.transform.Source;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.text.StrSubstitutor;
+import org.codice.ddf.catalog.ui.config.ConfigurationApplication;
+import org.codice.ddf.catalog.ui.forms.data.QueryTemplateMetacard;
+import org.codice.ddf.catalog.ui.forms.filter.FilterWriter;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.mockito.ArgumentCaptor;
+import org.opengis.filter.Filter;
+import spark.Spark;
+
+/**
+ * Spin up the {@link SearchFormsApplication} spark application and verify the REST contract against
+ * a mocked {@link CatalogFramework}.
+ *
+ * <p>This test suite uses HTTP to verify one of the data contracts for catalog-ui-search. By
+ * testing the network directly, the tests remain stable regardless of the JSON solution used.
+ *
+ * <p>This particular test suite verifies both directions of JSON and XML transforms using
+ * parameterized REST messages that are interpolated with actual test values at runtime. The
+ * objective of this test suite is to ensure serialization can handle any special character.
+ */
+@RunWith(Parameterized.class)
+public class SearchFormsSymbolsIT {
+  /**
+   * Set of symbols upon which to parameterize the tests; defined as they appear in a JSON document.
+   */
+  @Parameterized.Parameters(name = "Verify search form REST I/O for symbol: {0}")
+  public static Iterable<Object[]> data() {
+    return Arrays.asList(
+        new Object[][] {
```

Comment:
:question: Since each array only has one element, is there any reason not to just return an array and not a double array? https://github.com/junit-team/junit4/wiki/Parameterized-tests#tests-with-single-parameter